### ISSUE TITLE
Crash on PS3

### ISSUE THREAD

**Issue by Melluish** [igorciz777/RBC1GP-Translation-Pipeline#6]
When launching the resulting ISO on a modded PS3 you get an odd grey square on the right then the PS3 crashed to black before eventually going back to the XMB.

The original ISO is confirmed to work perfectly on PS3.

I used the patch intended for OPL.

Racing Battle - C1 Grand Prix (Japan).iso
MD5: 8131f1e1d205de32326d7c038f8f70a3

Racing Battle - C1 Grand Prix (Japan) (English Translation).iso
MD5: e6ba8abe6b4b8c5081ea26321b6f34a5

**Comment by igorciz777** [comment]
can you add more details about your ps3? is it a bc fat or just a slim, is it running hen or cfw etc etc

fixing this one might be tough because i only own a slim ps3 running hen, ill try to run it with the ps2 classics emu and see if i can replicate this

**Comment by Melluish** [comment]
Hi,

Certainly, it's a Slim. CFW 4.91 EvilNat. I have a BC fatty but can't break her out tonight. The game looks to start but just bugs as it's starting up. I thought it might be because the configs that get used for PS2 games aren't recognising it but I don't know how they're matching a config to the game, by the disc label?

**Comment by Melluish** [comment]
Does my checksum match what it should be after patching?

Racing Battle - C1 Grand Prix (Japan) (English Translation).iso
MD5: e6ba8abe6b4b8c5081ea26321b6f34a5

**Comment by igorciz777** [comment]
my checksum is a bit different
this is with the translation release 5, iso rebuilt with default imgburn settings and two different versions of gut archive tools, as the rebuilding method had few things changed

files patched with gut archive tools v3.1
MD5: 9A34CC66451584303870FBE96E8A7F62

files patched with gut archive tools v4.0
MD5: 07DAF4F70FFBBDDC39FDE4E6EBA8BC11

note that the hash can completely change with iso labels so its not the best way to confirm something is wrong

**Comment by Melluish** [comment]
Thank you for having a look at this. I'm in bed now but I'll check it out
tomorrow after work. I might've not patched it right.

**Comment by igorciz777** [comment]
Might be the case, game works fine on my end
I encrypted the iso with PS2 Classics GUI on pc

![456953271_913394717278001_6075043176211270830_n](https://github.com/user-attachments/assets/c79a3f5e-05b8-4f58-ac2b-5983b5f4749e)
![456606075_828612352755316_4308522033708488231_n](https://github.com/user-attachments/assets/5f7dc5cc-6d4c-4736-b99d-197f0e51f072)

**Comment by Melluish** [comment]
It's really strange, I can't get it to work :)

This is what the imgburn info pane shows me before I create the image. Does this look right?

![racing-battle-imgburn](https://github.com/user-attachments/assets/a7e5a47f-3ee4-4848-8aac-2569d2d6c22d)

**Comment by igorciz777** [comment]
I have a slight varation in size and sectors on my end 
![imgburn](https://github.com/user-attachments/assets/ffe63dce-68e2-4ad6-913d-9690342bf5e2)

just to confirm, the patching process in short:
1. run `gut_archive -r <your build.toc path> <your build.dat path> <in/ folder from downloaded release>`
2. remove the executable, copy paste the patched one, remove `_patchedElf` from the extension
3. copy paste the `MOV` folder and replace all files

I notice a non-windows ui on your screenshot, are you running imgburn through wine? maybe try using some natively running alternative for creating the iso

**Comment by Melluish** [comment]
Yeah I'm running the latest gut_archive Linux binary and ImgBurn through Wine. It's still doing the same thing. I have a Windows VM around here somewhere I'll try it tomorrow. Thanks man.

**Comment by Melluish** [comment]
God, I hate Windows hahaha
![Screenshot from 2024-09-05 01-32-51](https://github.com/user-attachments/assets/cd25a190-469a-46dc-a555-7bfea26265ac)

**Comment by Melluish** [comment]
God knows why ImgBurn through Wine is screwing up. I will try to just use `dd` at the weekend to see if I can get this to work without needing Windows.

**Comment by igorciz777** [comment]
This issue led me into the path of rebuilding the iso properly, apparently (and not surprising tbh) using imgburn is a very hacky way and it should only work for pcsx2 and not hardware 

Right now im playing around with the ps2sdk in a Windows XP VM :D

If i can find a way to rebuild the iso and keep a very similar structure to the original, i could make a simple one file xdelta patch to handle the entire translation, which would help out a ton of people with the installation process

**Comment by Melluish** [comment]
That would be awesome, mate. I was going to have a look this weekend to see if any of the Linux tools like `makeisofs` or `genisoimage` can produce an ISO that works. I think a single patch would be great. I suppose it's helpful that there's just the single release of the game. I did notice on Linux when I extract the contents of the disc the `IRX/` dir likes to get created in all lowercase, but `MOV/` and the files remain uppercase, whereas in Windows when I extract the disc all the folders and files are uppercase. Would you like me to close this as most people will be on Windows?

**Comment by igorciz777** [comment]
I tried to play around with `genisoimage` and `dd` with the proper flags, but it got rejected even by pcsx2

I dont want to discriminate against linux as im probably gonna move there after windows 10 eol myself, seeing how ridiculously bad everything related to win11 is

so lets keep this open in the case the single file patch idea doesn't work out

**Comment by Melluish** [comment]
No, that's fine, I love it but understand it's not for everyone. And sometimes things just work in Windows without issue.

I did make some discoveries tonight. So, even the checksum of the extracted files differed between Linux and Windows, so I knew I had to get that worked out first. I'm using `gCDEmu` to mount disc images and then just dragging them to a folder. It seems Linux likes to modify some of the file information when it mounts the image.

This can be fixed by using `7z` instead:

```
$ 7z x "Racing Battle - C1 Grand Prix (Japan).iso" -oextracted
```

The extracted files then match exactly to those extracted in Windows.

Then if you just apply the patch as per your instructions, using ImgBurn through Wine, the game works fine.

The final piece of the puzzle was trying to get `genisoimage` to output a working ISO. I couldn't get it to work, sadly. I have very limited knowledge of the layout of optical disc formats that I can't really look into this any more.

At least I can play this awesome game in English, thanks to you!
